taverna-cwl-activity/src/main/java/org/apache/taverna/cwl/CwlActivityFactory.java

```diff
@@ -0,0 +1,102 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.apache.taverna.cwl;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+import org.apache.taverna.cwl.utilities.CWLUtil;
+import org.apache.taverna.workflowmodel.Edits;
+import org.apache.taverna.workflowmodel.processor.activity.Activity;
+import org.apache.taverna.workflowmodel.processor.activity.ActivityConfigurationException;
+import org.apache.taverna.workflowmodel.processor.activity.ActivityFactory;
+import org.apache.taverna.workflowmodel.processor.activity.ActivityInputPort;
+import org.apache.taverna.workflowmodel.processor.activity.ActivityOutputPort;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class CwlActivityFactory implements ActivityFactory {
+	private static Logger logger = Logger.getLogger(CwlActivityFactory.class);
+	private static final int DEPTH_0 = 0;
+	private static final int DEPTH_1 = 1;
+	private Edits edits;
+
+	@Override
+	public Activity<?> createActivity() {
+		CwlDumyActivity activity = new CwlDumyActivity();
+		activity.setEdits(edits);
+		return activity;
+	}
+
+	@Override
+	public URI getActivityType() {
+		return null;
+	}
+
+	@Override
+	public JsonNode getActivityConfigurationSchema() {
+		return null;
```

Comment:
I'm not sure if this would pass with `null`. An example minimal schema which you could potentially adapt:

https://github.com/apache/incubator-taverna-plugin-gis/blob/master/taverna-gis-activity/src/main/resources/schema.json

returned as:

https://github.com/apache/incubator-taverna-plugin-gis/blob/master/taverna-gis-activity/src/main/java/org/apache/taverna/gis/GisActivityFactory.java#L55

```java

/*
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 *
 */
package org.apache.taverna.gis;

import static org.apache.taverna.gis.GisActivity.*;

import java.io.IOException;
import java.net.URI;
import java.util.HashSet;
import java.util.Set;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

import org.apache.taverna.workflowmodel.Edits;
import org.apache.taverna.workflowmodel.processor.activity.ActivityFactory;
import org.apache.taverna.workflowmodel.processor.activity.ActivityInputPort;
import org.apache.taverna.workflowmodel.processor.activity.ActivityOutputPort;

/**
 * Gis <code>ActivityFactory<code>.
 */
public class GisActivityFactory implements ActivityFactory {

	private Edits edits;

	@Override
	public GisActivity createActivity() {
		return new GisActivity();
	}

	@Override
	public URI getActivityType() {
		return URI.create(GisActivity.ACTIVITY_TYPE);
	}

	@Override
	public JsonNode getActivityConfigurationSchema() {
		ObjectMapper objectMapper = new ObjectMapper();
		try {
			return objectMapper.readTree(getClass().getResource("/schema.json"));
		} catch (IOException e) {
			return objectMapper.createObjectNode();
		}
}
```